src/vs/editor/browser/gpu/fullFileRenderStrategy.ts

```diff
@@ -0,0 +1,371 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { getActiveWindow } from '../../../base/browser/dom.js';
+import { Disposable } from '../../../base/common/lifecycle.js';
+import { EditorOption } from '../../common/config/editorOptions.js';
+import type { IViewLineTokens } from '../../common/tokens/lineTokens.js';
+import type { ViewportData } from '../../common/viewLayout/viewLinesViewportData.js';
+import type { ViewLineRenderingData } from '../../common/viewModel.js';
+import type { ViewContext } from '../../common/viewModel/viewContext.js';
+import type { ViewLineOptions } from '../viewParts/lines/viewLineOptions.js';
+import type { ITextureAtlasPageGlyph } from './atlas/atlas.js';
+import type { TextureAtlas } from './atlas/textureAtlas.js';
+import { BindingId, type IGpuRenderStrategy } from './gpu.js';
+import { GPULifecycle } from './gpuDisposable.js';
+import { quadVertices } from './gpuUtils.js';
+import { GlyphRasterizer } from './raster/glyphRasterizer.js';
+
+
+const enum Constants {
+	IndicesPerCell = 6,
+}
+
+const fullFileRenderStrategyWgsl = /*wgsl*/`
+struct GlyphInfo {
```

Comment:
I would love if we would extract this into its own `*.wgsl` file and have nice tooling for it (error checking, linting, syntax highlighting on GitHub etc).
If we have more such shaders this becomes increasingly more important.

Reply:
We could do that, however that means we cannot do the injection of shared words like `${BindingId.ViewportUniform}` and require manually ensuring they're in sync. The `/*wgsl*/` gives best of both worlds as it will give syntax highlighting with the WGSL Literal extension.

The way I'm thinking about it, we will probably only have 2-3 shaders total and their implementations will be hidden from most view parts